zebra-scan/src/service.rs

```diff
@@ -130,8 +131,33 @@ impl Service<Request> for ScanService {
                 .boxed();
             }
 
-            Request::Results(_key_hashes) => {
-                // TODO: read results from db
+            Request::Results(keys) => {
+                let db = self.db.clone();
+
+                return async move {
+                    if keys.len() > MAX_REQUEST_KEYS {
+                        return Err(format!(
+                            "maximum number of keys per request is {MAX_REQUEST_KEYS}"
+                        )
+                        .into());
+                    }
+
+                    let mut final_result = BTreeMap::new();
+                    for key in keys {
+                        let mut heights_and_transactions = BTreeMap::new();
+                        let txs = db.sapling_results_for_key(&key);
```

Comment:
We should call database methods from blocking threads to avoid blocking the asynx executor.

Reply:
Thanks for pointing to that. Fixed in https://github.com/ZcashFoundation/zebra/pull/8224/commits/80a429d727100ac3832031c97d3111e1b757ab4a